From 3365a4b59743aaf66e332c31dda131d853431039 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Sun, 16 May 2021 09:51:32 +0200
Subject: [PATCH 1/1] fs/squashfs: zero out unused fields in fs_dirent

When reading directories UEFI sub-system file attributes and timestamps are
required. Not all of these will be supplied by the squashfs. Zero out those
fields.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 fs/squashfs/sqfs.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fs/squashfs/sqfs.c b/fs/squashfs/sqfs.c
index 29805c3c6f..997be2dcf4 100644
--- a/fs/squashfs/sqfs.c
+++ b/fs/squashfs/sqfs.c
@@ -876,7 +876,7 @@ int sqfs_opendir(const char *filename, struct fs_dir_stream **dirsp)
 	char **token_list = NULL, *path = NULL;
 	u32 *pos_list = NULL;
 
-	dirs = malloc(sizeof(*dirs));
+	dirs = calloc(1, sizeof(*dirs));
 	if (!dirs)
 		return -EINVAL;
 
-- 
2.30.2

